src/backend/distributed/operations/stage_protocol.c

```diff
@@ -136,15 +136,14 @@ master_create_empty_shard(PG_FUNCTION_ARGS)
 		}
 	}
 
-	char partitionMethod = PartitionMethod(relationId);
-	if (partitionMethod == DISTRIBUTE_BY_HASH)
+	if (IsCitusTableType(relationId, HASH_DISTRIBUTED))
 	{
 		ereport(ERROR, (errmsg("relation \"%s\" is a hash partitioned table",
 							   relationName),
 						errdetail("We currently don't support creating shards "
 								  "on hash-partitioned tables")));
 	}
-	else if (partitionMethod == DISTRIBUTE_BY_NONE)
+	else if (IsCitusTableType(relationId, REFERENCE_TABLE))
```

Comment:
I guess we should  use `NO_DIST_KEY` as `master_create_empty_shard ` should error out for local tables.

(I don't care much about the error message, we can live with that)